packages/secure-store/secureUnlock.ts

```diff
@@ -0,0 +1,47 @@
+import { WalletUnlockError } from './error/WalletUnlockError'
+import { getSalt, storeSalt } from './secure-unlock/saltStore'
+import { deriveWalletKey, generateSalt } from './secure-unlock/walletKeyDerivation'
+import {
+  canUseBiometryBackedWalletKey,
+  getWalletKeyUsingBiometrics,
+  storeWalletKey,
+} from './secure-unlock/walletKeyStore'
+
+// TODO: how to version?
+const version = 1
+
+export { canUseBiometryBackedWalletKey }
```

Comment:
The functions, mainly their names, confuse me a bit and to me it is not really clear when to use which one. Also, since they are all exported, the API will get quite confusing I imagine.

Would it be possible to make it into an object/class with static methods, so we use it as a namespace?


i.e.

```js
// Wallet as a placeholder
Wallet.getSalt()
Wallet.createSaltAndIncrementVersion()
Wallet.deriveWalletKeyFromPin()
Wallet.unlockWallet() // do we need to mention using biometrics here?
```

Reply:
Yes we can do that, I'll do it when I integrate though, is that OK? Then I can play a bit more with how the API is easy to use. 

Re mentioning biometrics, I did want to mention it as there's a difference between unlocking the wallet using biometrics (no PIN/SALT needed) or using a PIN. As the secure store only deals with deriving and storing of wallet key, it doesn't actually open the wallet.

So it should be more "retrieveWalletKeyUsingBiometrics" and "deriveWalletKeyUsingPin", what do you think of these names?

Reply:
Names are good!